AngularJS/components/reports/reportCRUD.js

```diff
@@ -0,0 +1,141 @@
+function ReportListController(reportService) {
+    //Lists all reports in tabular form, with add, edit and remove options against each report item.
+    var $ctrl = this;
+    $ctrl.master = {};
+    $ctrl.report = {};
+    $ctrl.isUpdate = false;
+    $ctrl.current_report_item = {};
+    $ctrl.types = [
+        {
+            keys: "AN",
+            value: "Annual"
+        }, 
+        {
+            keys: "MO",
+            value: "Monthly"
+        }, 
+        {
+            keys: "WE",
+            value: "Weekly"
+        }
+    ];
+
+    $ctrl.$onInit(){
+        get_token();
+        get_reports();
+    }
+
+    function get_token(){
+        reportService.get_token().then(
+        //gets token from service to authenticate.
+            function() {
+                console.warn();
+            },
+            function(errorMessage) {
```

Comment:
error handler is repeating multiple times with same implementations. can be written a named function as reused in all ajax calls.